organization_token.go

```diff
@@ -53,6 +67,35 @@ type OrganizationTokenCreateOptions struct {
 	// Optional: The token's expiration date.
 	// This feature is available in TFE release v202305-1 and later
 	ExpiredAt *time.Time `jsonapi:"attr,expired-at,iso8601,omitempty"`
+	// Optional: What type of token to create
+	// This option is only applicable to HCP Terraform and is ignored by TFE.
+	TokenType *TokenType
+}
+
+// OrganizationTokenReadOptions contains the options for reading an organization token.
+type OrganizationTokenReadOptions struct {
+	// Optional: What type of token to read
+	// This option is only applicable to HCP Terraform and is ignored by TFE.
+	TokenType *TokenType
+}
+
+// OrganizationTokenDeleteOptions contains the options for deleting an organization token.
+type OrganizationTokenDeleteOptions struct {
+	// Optional: What type of token to delete
+	// This option is only applicable to HCP Terraform and is ignored by TFE.
+	TokenType *TokenType
+}
+
+// Creates the URL for use against organization token API routes
+func (s *organizationTokens) generateURL(organization string, tokenType *TokenType) string {
+	u := fmt.Sprintf("organizations/%s/authentication-token", url.QueryEscape(organization))
```

Comment:
url.PathEscape

Reply:
I'm happy to change it (and it should be) but as an FYI, QueryEscape is littered throughout the codebase :-(

Reply:
Fixed